packages/autocomplete-preset-algolia/src/search/fetchAlgoliaResults.ts

```diff
@@ -35,6 +35,11 @@ export function fetchAlgoliaResults<TRecord>({
 }: SearchParams): Promise<
   Array<SearchResponse<TRecord> | SearchForFacetValuesResponse>
 > {
+  if (typeof searchClient !== 'object') {
+    throw new Error(
+      'The `searchClient` option must be an initialized search client.'
```

Comment:
If this happens that probably means that they didn't set the option at all. Should we give more context so that they would know where to set it?

Reply:
the problem is that this function is used for both facets and hits, so we can't say "to getAlgoliaResults". Do you have something in mind?

A common mistake is doing:

```
const customClient = ...

getAlgoliaResults({ customClient })
```

Reply:
I see. Alternatively, maybe the check can be done on a higher level, directly in `getAlgoliaResults` / `getAlgoliaFacets`, so that the information provided can be more easily contextualized, at the cost of duplicating the code.

In any case, I'm okay with this as is.

Reply:
I agree that the error message is confusing. If the common error is that they pass a wrong object key, they're likely not comfortable with the language, and passing an "initialized search client" doesn't help them understanding what's going wrong.